### DISCUSSION TITLE
Endless 'Waiting for a runner to pick up this job...'

### DISCUSSION THREAD

**Discussion by stukalo** [community/community#31587] [npm] [answered]
Hi, folks!
I'm trying to setup a workflows, but the runner doesn't start my job. I've requested `ubuntu-latest` and got a message `Waiting for a runner to pick up this job...`. I've canceled jobs and triggered it again for several times without success.
Could anyone give me a hint to what I should do in this case?

Also, it seems that time shows wrong value on UI
[![image](https://user-images.githubusercontent.com/25886788/187868472-65473710-9e94-4d53-8065-21c0fd83b03d.png)](url)

**Comment by airtower-luna** [comment, marked as answer]
The message looks like there might be a typo in the `runs-on` definition. But the wrong time indicates there might be another bug. If the workflow file is correct or you still have the same issue after fixing it, the best option is probably to [submit a bug report](https://support.github.com/contact/bug-report).

**Reply by stukalo** [reply]
Thank you!

**Reply by MikeJansen** [reply]
Just ran into this myself.  Was putting `ubuntu:latest`...

**Reply by caijianhong** [reply]
Thank you!

**Reply by swiknaba** [reply]
There is actually a quirk with how the YML file is parsed, and the order of some keys matters, see https://github.com/orgs/community/discussions/31587#discussioncomment-9893422

**Reply by HarimbolaSantatra** [reply]
Thanks! I typed `ubuntu:22` instead of `ubuntu-22.04`.

**Reply by Austin1serb** [reply]
why are workflow .yaml files so damn hard to work with. worst developer experience i have had so far with any language

**Reply by Zenitugo** [reply]
Thank you so much

**Reply by mirabilos** [reply]
same here for `ubuntu-latest`, and the ticket got immediately closed (the worklfow definition was not even changed from the previous runs).

https://github.com/mirabilos/dygraphs/actions/runs/34738482151/workflow

**Comment by quddlr9015** [comment]
I think there is a typo. You typed ubuntu-lates. It should be ubuntu-latest

**Comment by tueda** [comment]
I had a similar issue with `ubuntu-20.4` and struggled for a half day...

**Reply by ghost** [reply]
You must use "ubuntu-20.04" instead "ubuntu-20.4", that is your mistake

**Reply by tueda** [reply]
You are right. Then, the point is: if GitHub Actions could instantly give me an error with some adequate message, I didn't spend a half day.

**Comment by salamisodikiolawale** [comment]
Hello !, i have the same problem but for me : 
When I launch my ci everything works fine with runs-on: ubuntu-latest but my runners does not detect

![image](https://user-images.githubusercontent.com/28807604/215774852-bd529e41-fe7b-4977-be57-4b5ba315685b.png)

```
name: CI-CD

on:
  push:
    branches: [ "develop" ]
  pull_request:
    branches: [ "develop" ]

jobs:

  my-job:
      runs-on: ubuntu-latest
      defaults:
        run:
          working-directory: ./frontend

      strategy:
        matrix:
          node-version: [16.x]

      steps:
      - name: CHECK-OUT GIT REPOSITORY
        uses: actions/checkout@v3
      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v3
        with:
          node-version: ${{ matrix.node-version }}
          cache: 'npm'
          cache-dependency-path: '**/package-lock.json'
      - name: INSTALL DEPENDENCIES
        run: npm ci
      - name: BUILDING APP
        run: npm run build --if-present
      #- name: TESTINT APP
      #- run: npm test
      - name: Build & push Docker image
        uses: mr-smithers-excellent/docker-build-push@v5
        with:
        
          image: user/frontend    
          tags: latest
          registry: docker.io
          dockerfile: frontend/Dockerfile
          username: ${{ secrets.DOCKER_USERNAME }}
          password: ${{ secrets.DOCKER_PASSWORD }}


```

**Comment by kojix2** [comment]
I didn't realize until this page that I was using `ubuntu` instead of `ubuntu-latest`. Saved me a lot of time. Thank you very much.

**Comment by tueda** [comment]
Eventually, now `ubuntu-18.04` ([deprecated](https://github.com/actions/runner-images/issues/6002)) enters the eternal waiting. It worked in March.

**Reply by airtower-luna** [reply]
That's because it isn't just deprecated any more, it's been removed. So any workflow using it can't ever get a runner. Update to a newer version, `ubuntu-22.04` is the current one.

**Reply by airtower-luna** [reply]
See: https://docs.github.com/en/actions/using-github-hosted-runners/using-larger-runners

**Reply by jtormey** [reply]
I just ran into this as well, with `ubuntu-18.04`. Would definitely prefer GitHub Actions to return an error if the `runs-on` config specifies an unsupported platform.

**Reply by pdfrod** [reply]
Just ran into it as well; my team was completely puzzled as to why the workflows were not running. A better error message would make a huge difference here.

**Comment by joeflack4** [comment]
I am using large runners. My problem was that I just created the runner, and it auto-assigns it to a group called "Default". But by default, all runners in this group do not allow you to run actions on _public repositories_  unless you click the check box for "**Allow public repositories**".

**Reply by joeflack4** [reply]
Could we improve the UX on this?

If running action on public runner w/ no permissions, can we exit out and show an error message about such insufficient permissions? Or even better, deactivate the action button, preferably with a tooltip explaining why.

**Reply by foker** [reply]
might be better to move into a separated issue

**Comment by sahilrajput03** [comment]
For me I just stopped the worflow and started it again and it started working now. I think for the first push you need to do it twice to make it work! Thanks.

**Reply by sujitpursuit** [reply]
thanks! restarting did make it work for me too!

**Reply by VijayAllRounder** [reply]
it worked for me too,thanks

**Reply by tomasapica** [reply]
life saver

**Comment by jenrenia** [comment]
I couldn't run it until I added one blank line after the `jobs:`(Does anyone know if this really matters?)
<img width="356" alt="image" src="https://github.com/community/community/assets/22729123/e7b11ed6-5a8d-48c1-9a00-e95751d4879c">

**Comment by Maclenn77** [comment]
I'm experiencing the same issue, I have tried and review some of the things pointed out here, but still not working. I'm using an aws auto-generated template for lambdas, so I'm wondering if the issue it's one of "uses" file that are being referenced in the job. 

So, this is the part with issues:

```yaml
 build-and-package:
    if: github.ref == 'refs/heads/main'
    needs: [test]
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: aws-actions/setup-sam@v2
        with:
          use-installer: true

      - name: Build resources
        run: sam build --template ${SAM_TEMPLATE} --use-container
```

In this part, the action is stuck:

```bash
Building codeuri: /home/runner/work/about_me/about_me/about_me runtime: ruby3.2 metadata: {} architecture: arm64 functions: AboutMeFunction
Fetching public.ecr.aws/sam/build-ruby3.2:latest-arm64 Docker container image........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
Mounting /home/runner/work/about_me/about_me/about_me as /tmp/samcli/source:ro,delegated, inside runtime container

```



When I check the logs, these are the messages:

```
2024-03-04T03:46:57.9991600Z Requested labels: ubuntu-latest
2024-03-04T03:46:57.9991913Z Job defined at: Maclenn77/about_me/.github/workflows/pipeline.yaml@refs/heads/main
2024-03-04T03:46:57.9992032Z Waiting for a runner to pick up this job...
2024-03-04T03:46:58.8924246Z Job is waiting for a hosted runner to come online.
2024-03-04T03:47:01.3614880Z Job is about to start running on the hosted runner: GitHub Actions 13 (hosted)
```

Not sure if it's related, but in the summary there is a warning:

```
TEST
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/checkout@v3. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.
```

I don't think so, as it seems that is a warning triggered by the previous job.

As other people pointed out, there's no hint about what could cause the problem. I'm new using Github Actions for CI/CD with AWS. This pipeline is for testing the configuration after following the docs of AWS and Github. I'm shooting in the dark right now, I don't know if the problem are permissions, configuration or anything else.
 
I would be thankful with anyone that could figure out what is causing the problem and also support people that pointed out that better error messages would be do a big difference.

--

**Comment by 4Source** [comment]
I have a similar problem. I would like to run a job on ubuntu-latest, windows-latest, macos-latest. When I enter them individually everything starts but when I enter them together it just says "Waiting for a runner to pick up this job..."

```
name: Continuous Integration

on: 
  pull_request:
    branches:
      - master
    # paths: 
    #   - 'src/**'
  push: 
    branches:
      - master
    # paths: 
    #   - 'src/**'
  workflow_dispatch:

jobs:
  test:
    name: Run unit tests
    runs-on: [ubuntu-latest, windows-latest, macos-latest]

    steps:
      - name: Checkout repository 
        uses: actions/checkout@v4

      - name: Use Node.js "18.x"
        uses: actions/setup-node@v4
        with:
          node-version: "18.x"
      
      - name: Install Dependencies
        run: npm ci

      - name: Run Tests
        run: npm run test
```

![image](https://github.com/community/community/assets/38220764/824633f2-9881-4a18-b9c0-dc9793994c94)

**Comment by swiknaba** [comment]
Got the same issue, and my syntax and all defined values were valid. 

**Solution**: remove new lines

in _some_ places, new lines seem to break Github Actions completely. 

I had to remove a new line after:

**Good**:
```
jobs:
  test:
```

**Bad**:
```
jobs:

  test:
```

as well as 

**Good**:
```
jobs:
  test:
    strategy: xx
    runs-on: ubuntu-latest
```

**Bad**:
```
jobs:
  test:
    strategy: xx

    runs-on: ubuntu-latest
```

The VSCode Actions Plugin did not catch this as a validation error.

**Comment by Bulat-Ziganshin** [comment]
@swiknaba 
may be, it adds "\n" to the string value? I found this problem when making a long string with ": >" syntax, and finally solved it with ": >-" variation, although YAML docs say it's not necessary

**Comment by MilesSpence** [comment]
For anyone else having issues with this I think the reason mine was not working was because I was on a GitHub Enterprise Server (GHES) and per this page: [https://docs.github.com/en/enterprise-server@3.9/actions/hosting-your-own-runners/managing-self-hosted-runners](https://docs.github.com/en/enterprise-server@3.9/actions/hosting-your-own-runners/managing-self-hosted-runners) it says, "Note: GitHub-hosted runners are not currently supported on GitHub Enterprise Server. You can see more information about planned future support on the [GitHub public roadmap](https://github.com/github/roadmap/issues/72)." So I think you have to host your own runner.

**Reply by annedroiid** [reply]
Thank you, this was the issue for me.

**Comment by gilbitron** [comment]
For anyone else who ends up here, the issue for me was using custom labels for runners, which is now deprecated.

![CleanShot 2024-06-27 at 11  01 17 2@2x](https://github.com/community/community/assets/203882/44be598b-ac55-4025-875f-48a22be1f1ef)

**Reply by ryeleo** [reply]
Are labels actually deprecated? I don't see any mention of this in the [GHES 3.15 docs](https://docs.github.com/en/enterprise-server@3.15/actions/hosting-your-own-runners/managing-self-hosted-runners/using-labels-with-self-hosted-runners)

**Comment by swiknaba** [comment]
I've reached GitHub support and got the message attached below.

TL;DR: **The order of _some_ keys in the YML file actually matters.**


=====

Thanks for contacting GitHub Support.
 
So, while the syntax is correct, **the ordering is a little different** to what we show in the [guidelines in our documentation](https://docs.github.com/en/actions/quickstart). The general convention when writing your yml files is to specify the type of runner for the job at the top level of each job in the workflow, directly under the job name. This means that the start of the job definition would look like:

```
jobs:
  azure:
    runs-on: ubuntu-latest
```

I agree that it's strange that one version works if you simply add a space, and it's also strange that the VSCode Github Actions Extension didn't show a validation error; however, this relates to another convention used in yml files: the - uses key is generally defined after a step name, so the definition of your steps should look something like the following:

```
steps:
  - name: Checkout code
    uses: actions/checkout@v4
  - name: init
    uses: opentofu/setup-opentofu@v1
    run: _do-some-magic_
```

=====

NOTE: re-ordering the `uses`-key solved the problem for me.

**Comment by sarnold** [comment]
We hit this today; turned out to be a silly/horrible typo in the compiler name in a matrix build.  Better yaml checks *and* error message would definitely help save time and frustration...

**Comment by p69** [comment]
My workflow eventually stoped executing and hanging forever in queued state. I even haven't changed anything. I have no idea how to fix it.
Here is my yaml
```yaml
name: Deploy to DigitalOcean
on:
  push:
    branches:
      - dev
jobs:
  deploy:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout code
        uses: actions/checkout@v4
      - name: Set up SSH
        uses: webfactory/ssh-agent@v0.9.0
        with:
          ssh-private-key: ${{ secrets.DO_SSH_PRIVATE_KEY }}
      - name: Add DigitalOcean to known hosts
        run: |
          mkdir -p ~/.ssh
          ssh-keyscan -H ${{ secrets.DO_IP_ADDRESS }} >> ~/.ssh/known_hosts
      - name: Sync project files to server
        run: |
          rsync -avz --exclude 'deps' --exclude '_build' ./ ${{ secrets.DO_USER }@${{ secrets.DO_IP_ADDRESS }}:/${{ secrets.DO_USER }/apps/clicker
      - name: Set execute permissions on the script
        run: |
          ssh root@${{ secrets.DO_IP_ADDRESS }} 'chmod +x /${{ secrets.DO_USER }/apps/clicker/build_and_run_dev.sh /${{ secrets.DO_USER }/apps/clicker/docker/run_local_dev.sh'
      - name: Run the scripts
        run: |
          ssh ${{ secrets.DO_USER }@${{ secrets.DO_IP_ADDRESS }} 'cd /${{ secrets.DO_USER }/apps/clicker && ./build_and_run_dev.sh'

```

**Reply by iagonuvem** [reply]
Eventually my workflow stopped working, I've just done a deployment early today... not sure what happened, commenting to watch the thread.

**Reply by messenger1976** [reply]
Same here.

**Reply by a94763075** [reply]
it seems like github's [issues](https://www.githubstatus.com/incidents/m70hk23gx3nx)

<img width="1029" alt="image" src="https://github.com/user-attachments/assets/d8ffafb6-549c-4ba5-8904-b91bb2dcc254">

**Reply by p69** [reply]
Oh, then it's not just me.

**Reply by willTastyMakers** [reply]
Nope,  I have been trying to get out a release for 3 hours

**Reply by gauravbordoloi** [reply]
Me too :/

**Reply by pollux-** [reply]
Same here as well 👎

**Reply by Deepanshusuman** [reply]
ahh waiting guys

**Reply by wenchaoliao** [reply]
+1

**Reply by a94763075** [reply]
It resolved, start working  
<img width="517" alt="image" src="https://github.com/user-attachments/assets/45ce6e8c-4cf5-441a-9596-f10d7b536841">

<img width="1112" alt="image" src="https://github.com/user-attachments/assets/2e671ed1-e984-4602-a6a1-e66d286afa57">

**Reply by p69** [reply]
Yeah, but it seems unstable. My action fails with some random reasons

**Reply by uzairash** [reply]
same issue didn't change a thing in workflow file, it was working fine but now getting stuck

**Reply by chandravirbytesfarms** [reply]
getting the same issue pipeline getting stuck in queue state

**Reply by minhchienwikipedia** [reply]
The issue back!

**Reply by Tchangang** [reply]
Same problem. help please. production impacted !

**Reply by minhpq-vbee** [reply]
Same here.

**Reply by gu3st** [reply]
Same issue for us!

**Reply by thiraphat-ps-dev** [reply]
me too !

**Reply by jeeeepp** [reply]
Same here. The issue back.

**Reply by Teja-Cuber** [reply]
I guess it has been resolved, it started working.

**Reply by Map987** [reply]
it worked

**Comment by myanpetra99** [comment]
well i currently have this issue

**Comment by MetaCrawler** [comment]
Aaand today it started for us.
No matter which ubuntu version we use.
Fails with latest as well as 22.04, 23.04. 24.04.

**Comment by matt335672** [comment]
We've now encountered this too. We're not using self-hosted runners.

I implemented the changes in @swiknaba's [post](#discussioncomment-9893422). These worked fine for me in my account, but don't work when I push the changes to the project account.

**Update:** As of 2024-08-06 09:45 UTC everything here seems to be working again with our original YAML files.

**Comment by rym-dd** [comment]
Same here, all pipelines running on GitHub runners (not self-hosted) are stuck.

**Reply by ArnavWee** [reply]
Did it resolve itself or did you do something to fix it , facing the same issue

**Reply by dangdong163** [reply]
HI @ArnavWee i facing that issue currently. don't know why. it's ok for last 1 hour

**Reply by klonwar** [reply]
This problem appears and disappears for no reason. Github support has not helped so far :c

**Comment by klonwar** [comment]
Same for me with self hosted on ubuntu... It started not so long time ago, before it worked like a charm

`sudo ./svc status` shows `Active: active (running)`
In Settings -> Actions -> Runners it shows `Status: Idle`

But in action log `Waiting for a runner to pick up this job...`

Enabling and disabling a label in the github ui helps for a little time
How to fix this?

**Reply by john-from-qa** [reply]
This is definitely an annoying issue.  I constantly have to cancel jobs and resubmit them, then it works fine.

**Comment by dev-javid** [comment]
In case your repository is public, make sure you allow access explicitly:

![image](https://github.com/user-attachments/assets/7754d45f-43c9-45eb-a455-9fc24b2d3074)

**Comment by dfernandezramos** [comment]
Is anybody experiencing this issue with self-hosted runners now? All I can see is that the CI gets stuck forever waiting the runner to pick the second job of the workflow until a new workflow is created. Then the old one gets unstuck, finishes, the new workflow gets stuck again on the second job and so on. In a nutshell, we need to trigger a new workflow to make the previous one pick the remaining job and finish.

It seems it only happens with our Windows self-hosted node. It does not happen with our self-hosted macOS node.

Our yaml files are exactly the same. This has never happened to us until yesterday.

**Reply by melyux** [reply]
Happens to my macOS self-hosted runners.

**Reply by berkaycit-rotatelab** [reply]
did you find any solution?

**Reply by dfernandezramos** [reply]
@berkaycit-rotatelab the very next day it started working again by itself ¯\_(ツ)_/¯
Never happened again, so I guess it was a Github issue...

**Reply by sglbl** [reply]
I had this error on self-hosted when I added needs (multiple jobs)

**Comment by tzing0013** [comment]
![image](https://github.com/user-attachments/assets/31f3e488-7291-45c3-b26c-b410aa779c11)
![image](https://github.com/user-attachments/assets/2c76c334-b8da-424d-ad4f-78b3a77b57e2)
same to  me, and it request a label i don't  have, after add the label,  it works

**Comment by micksatana** [comment]
Experienced the same issue here. But, in my case, simply cancel the workflow and re-run all jobs work right away.

I was using `ubuntu-latest`

**Comment by chr4ss12** [comment]
guys this whole thing is EXTREMELY fiddely, can you put some resources to this? I had no luck getting github hosted runner to work after 2 hours of messing around

**Comment by gfviegas** [comment]
After this issue popped out again today, there was a issue in Github servers that I could find out more info on https://www.githubstatus.com/ 

If this issue happens to you, I suggest visiting the status page to see if there is a problem on the Github servers.

**Reply by ams-amoin-telus** [reply]
There's an outage going on with GH Action runners. Self-hosted runners are not impacted.
